table/sstable/iterator.go

```diff
@@ -292,12 +295,28 @@ func (itr *Iterator) seekBlock(key []byte) int {
 	})
 }
 
+func (itr *Iterator) seekPlr(key []byte) int {
+	k, _ := strconv.Atoi(string(key))
+	blk, _ := itr.plr.predict(float64(k))
+	lower, upper := int(blk - 1.0), int(blk + 1.0)
+	return sort.Search(upper - lower, func(idx int) bool {
+		blockBaseKey := itr.tIdx.baseKeys.getEntry(lower+idx)
+		return bytes.Compare(blockBaseKey, key) > 0
+	})
+}
+
```

Comment:
Looks like you used the error bound of only 1.0? That might create too many segments. I guess we can start trying it with 8.0.

Also to handle the lower value you would need to call `max(int(blk - 8.0), 0)` and for the upper as well as `min(int(blk + 8.0), TABLE_BLOCK_SIZE - 1)`

Let me try make a PR into your branch

Reply:
> Let me try make a PR into your branch

I was just too sleepy then and  wrote a hard code here to make this piece of code a little more complete ......   :p,